### DISCUSSION TITLE
How do I use ESM modules for Serverless Functions in Typescript?

### DISCUSSION THREAD

**Discussion by debater-coder** [vercel/community#1090] [Help] [answered]
Hi there,

I've been using ESM modules in my Serverless Functions for a while now by adding the `.mjs` extension. Recently, I've been migrating my project to TypeScript, and I haven't been able to find a way to use ESM in TypeScript. I have tried using `type: module` in `package.json` but that does not seem to work. Also, the `.mts` extension is unsupported for serverless functions. Is there any way to use ESM modules in Serverless Functions with TypeScript.

Thanks!

**Comment by amyegan** [comment, marked as answer]
Hi @debater-coder! There is an open issue (https://github.com/vercel/vercel/issues/7908) and a draft PR (https://github.com/vercel/vercel/pull/7954) in progress to address this scenario
